scene/2d/physics/animatable_body_2d.cpp

```diff
@@ -87,9 +89,9 @@ void AnimatableBody2D::_notification(int p_what) {
 
 		case NOTIFICATION_LOCAL_TRANSFORM_CHANGED: {
 			// Used by sync to physics, send the new transform to the physics...
-			Transform2D new_transform = get_global_transform();
+			next_transform = next_transform * last_valid_transform.affine_inverse() * get_global_transform();
```

Comment:
In my uttermost ignorance of the topic, why... why does this work?

Reply:
Essentially, it adds the difference.
```
diff = global - last_valid
next = next + diff
```

Shorthand
```
next += global - last_valid
```

Could also write it like this
```
next += (-last_valid) + global
```

or this
```
next += inverse(last_valid) + global
```

But these are transform matrix, need to combine them with a special * operator.
```
next = next * inverse(last_valid) * global
```